enhancements/sig-architecture/63-hosted-addon/README.md

```diff
@@ -0,0 +1,262 @@
+functionality is not affected.
+
+#### Story 2
+
+As a hub cluster administrator, on the premise that the addon functionality is not affected, I do not want the addon 
+agent to talk to the hub cluster directly. 
+
+### Terms
+
+- `Hosted` mode: deploy addons agents deployments outside the managed cluster.
+- `Hosting cluster`: The cluster where the addons agents deployments runn.
+
+### Constraints
+
+There are several constraints:
+- The managed cluster must be exposed to the hosting cluster
+- The managed cluster(Klusterlet) must be imported to hub in `Hosted` mode
+- The hosting cluster must be a managed cluster of the hub, hosting cluster should be managed by the hub
+- The hosting cluster must be the same cluster where the managed cluster klusterlet(registration-agent & work-agent) 
+runs
+- After the addon is deployed, the deployment mode cannot be switched
+
+## Design Details
+
+### For addon developers
+
+We propose to add a label `addon.open-cluster-management.io/manifest-type` for the addon agent manifests to mark which 
+manifests should be deployed where in Hosted mode:
+- No matter in Default mode or Hosted mode, all manifests will be deployed on the managed cluster
+
+- When the label does not exist or its value is `hosted-mode-managed`: the manifest will be deployed on the managed 
+cluster in Hosted mode
+- When the label's value is `hosted-mode-hosting`: the manifest will be deployed on the hosting cluster in Hosted mode
+- When the label's value is `hosted-mode-none`: the manifest will not be deployed in Hosted mode
+
+
+<table class="tg">
+<thead>
+  <tr>
+    <th class="tg-0pky" rowspan="2"><br>Label Value</th>
+    <th class="tg-c3ow">Default Mode</th>
+    <th class="tg-c3ow" colspan="2">Hosted Mode</th>
+    <th class="tg-c3ow" rowspan="2"><br>Example</th>
+  </tr>
+  <tr>
+    <th class="tg-c3ow">Managed Cluster</th>
+    <th class="tg-c3ow">Managed Cluster</th>
+    <th class="tg-c3ow">Hosting Cluster</th>
+  </tr>
+</thead>
+<tbody>
+  <tr>
+    <td class="tg-0pky"><span style="font-weight:400;font-style:normal">nonexist | hosted-mode-managed</span></td>
+    <td class="tg-c3ow">Y</td>
+    <td class="tg-c3ow">Y</td>
+    <td class="tg-c3ow"></td>
+    <td class="tg-c3ow">CRD owned by the addon</td>
+  </tr>
+  <tr>
+    <td class="tg-0pky"><span style="font-weight:400;font-style:normal">hosted-mode-hosting</span></td>
+    <td class="tg-c3ow">Y</td>
+    <td class="tg-c3ow"></td>
+    <td class="tg-c3ow">Y</td>
+    <td class="tg-c3ow">deployment</td>
+  </tr>
+  <tr>
+    <td class="tg-0pky"><span style="font-weight:400;font-style:normal">hosted-mode-none</span></td>
+    <td class="tg-c3ow">Y</td>
+    <td class="tg-c3ow"></td>
+    <td class="tg-c3ow"></td>
+    <td class="tg-c3ow"></td>
+  </tr>
+</tbody>
+</table>
+
+### For addon deployers
+
+Once the managed cluster is imported to the hub in `Hosted` mode, we propose to add an annotation 
```

Comment:
Is the annotation something that would be added automatically when the managedcluster is in Hosted mode? Or would this be specified by the user / another process when creating the ManagedClusterAddOn resource?

Reply:
The annotation will not be added automatically, it should be specified by the user / another process when creating the ManagedClusterAddOn resource. since even the managedcluster is in Hosted mode, the addon still can be in Default mode.